### ISSUE TITLE
Issues with datastore

### ISSUE THREAD

**Issue by zyounker** [maintux/knife-esx#19]
When i specify the datastore i receive the following error:

knife esx vm create --esx-host 10.10.30.54 --vm-name "test-vm" --datastore "PHXESX117 LOCAL DATASTORE SSD" --use-template "Ubuntu_14.04_LTS_Server_Template_v1.vmdk" --esx-templates-dir "/vmfs/volumes/iso/Ubuntu_14.04_LTS_Server_Template_v1"
Connecting to ESX host 10.10.30.54... 
Creating VM test-vm
Using template Ubuntu_14.04_LTS_Server_Template_v1.vmdk
Cloning template...
ERROR: RbVmomi::Fault: FileNotFound: File [PHXESX117 LOCAL DATASTORE SSD] test-vm/test-vm.vmdk was not found

I have tried escaping and quoting the spaces without much luck. it doesn't actually create the cloned VM in the datastore. so i tried to use the datastore ID:

knife esx vm create --esx-host 10.10.30.54 --vm-name "test-vm" --datastore 53d10a09-52095828-89dd-000af714d6c0 --use-template "Ubuntu_14.04_LTS_Server_Template_v1.vmdk" --esx-templates-dir "/vmfs/volumes/iso/Ubuntu_14.04_LTS_Server_Template_v1"
Connecting to ESX host 10.10.30.54... 
Creating VM test-vm
Using template Ubuntu_14.04_LTS_Server_Template_v1.vmdk
Cloning template...
ERROR: RbVmomi::Fault: InvalidDatastore: Invalid datastore format '53d10a09-52095828-89dd-000af714d6c0'.

So the 2nd command actually creates the copy of the VM in the data store, but then fails as it says the datastore name is invalid.

It seems if i run the 2nd command and then the first. I am able to get it to create and power on the VM as a workaround.. Any ideas why it doesn't seem to work with either datastore name?

**Comment by djdefi** [comment]
Does your datastore actually have spaces in the name?

Was just looking at my notes and your command looks very similar to mine, however we do not use spaces in the names of our data stores.

**Comment by zyounker** [comment]
Yes, The DS has spaces in the name. I'm working with existing resources that are not going to be easy to change. Which is why i was trying the datastore id 53d10a09-52095828-89dd-000af714d6c0.

I have also tried quoting the datastore name with spaces as well as quoting and escaping the spaces.  By escaping the spaces i can get it to clone the VM, but then it fails with invalid datastore format. If i run it again without escaping the quotes it works. (since the VM clone already exists).

**Comment by djdefi** [comment]
Have you tried escaping the spaces with a backslash? Like:
 DATA\ STORE\ NAME

Not sure if that will work just a guess

**Comment by zyounker** [comment]
Yes, tried that.. When doing that it creates a clone, but then fails with invalid datastore. Using a combination of both escaped and non-escaped gets it to progress further. The first time i can use escaped spaces which creates the clone and fails with invalid datastore. Then i run it again without escaped spaces and it works.

If i run it without escaped spaces the 1st time, it fails with FileNotFound error and does not create a clone on the volume.
